packages/core/admin/admin/src/content-manager/utils/mergeMetasWithSchema.js

```diff
@@ -1,12 +1,13 @@
 import set from 'lodash/set';
+import merge from 'lodash/merge';
 
 const mergeMetasWithSchema = (data, schemas, mainSchemaKey) => {
   const findSchema = (refUid) => schemas.find((obj) => obj.uid === refUid);
   const merged = Object.assign({}, data);
   const mainUID = data[mainSchemaKey].uid;
   const mainSchema = findSchema(mainUID);
 
-  set(merged, [mainSchemaKey], { ...data[mainSchemaKey], ...mainSchema });
+  set(merged, [mainSchemaKey], merge({}, mainSchema, data[mainSchemaKey]));
```

Comment:
I know there is a project to remove Lodash from front. But I don't want to recode a merge function. Do we have any that doesn't use Lodash?

Reply:
Because the déstructuration of object only merge the first layer of objects. If there is nested object, the second one will overwrite the first one.

Reply:
Okay, and just to be clear, we're expecting nested data structures in this object? In which case, maybe lets just add a `TODO:` note above so we're not adding in this code without acknowledging we want to change it.